application/models/services/SurveyUpdater.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace LimeSurvey\Models\Services;
+
+use LimeSurvey\Models\Services\SurveyUpdater\{
+    LanguageSettings,
+    GeneralSettings,
+    UrlParams,
+    TemplateConfiguration
+};
+use LimeSurvey\Models\Services\Proxy\ProxyExpressionManager;
+use LimeSurvey\Models\Services\Exception\{
+    ExceptionPersistError,
+    ExceptionNotFound,
+    ExceptionPermissionDenied
+};
+
+/**
+ * Survey Updater Service
+ *
+ * Service class for update survey settings.
+ *
+ * Dependencies are injected to enable mocking.
+ */
+class SurveyUpdater
+{
+    private LanguageSettings $languageSettings;
+    private GeneralSettings $generalSettings;
+    private UrlParams $urlParams;
+    private ProxyExpressionManager $proxyExpressionManager;
+    private TemplateConfiguration $templateConfiguration;
+
+    public function __construct(
+        LanguageSettings $languageSettings,
+        GeneralSettings $generalSettings,
+        UrlParams $urlParams,
+        ProxyExpressionManager $proxyExpressionManager,
+        TemplateConfiguration $templateConfiguration
+    ) {
+        $this->languageSettings = $languageSettings;
+        $this->generalSettings = $generalSettings;
+        $this->urlParams = $urlParams;
+        $this->proxyExpressionManager = $proxyExpressionManager;
+        $this->templateConfiguration = $templateConfiguration;
+    }
+
+    /**
+     * Update
+     *
+     * @param int $surveyId
+     * @param array $input
+     * @throws ExceptionPersistError
+     * @throws ExceptionNotFound
+     * @throws ExceptionPermissionDenied
+     * @return array
+     */
+    public function update($surveyId, $input)
+    {
+        $this->languageSettings->update(
+            $surveyId,
+            $input
+        );
+
+        $meta = $this->generalSettings->update(
```

Comment:
If you look at the original methods from database controller "actionUpdateSurveyLocaleSettingsGeneralSettings" and "actionUpdateSurveyLocaleSettings" they both update some survey object settings and also language specific settings in a single request. I refactored this into a SurveyUpdated::update(). Only data that you pass in is actually updated. If you don't pass in language specific settings we don't attempt yo update anything language specific.

The focus is refactoring existing functionality for re-usability and at the same time making the code testable. This may not be the best way to organise the code but generally improving code organisation was not the focus of task. 

The more "issues" we try to fix in one go, the more code we change, the higher the chance of us breaking something. 

![Screenshot 2023-06-27 at 12 01 45](https://github.com/LimeSurvey/LimeSurvey/assets/2025743/3d4b3743-4ffb-46c8-88e0-59e968efcb03)

Reply:
Alright all good :)